include/dynamatic/Dialect/Handshake/HandshakeOps.td

```diff
@@ -983,22 +983,29 @@ def SpeculatorOp : Handshake_Op<"speculator", [
 
     ```mlir
     %dataOut, %saveCtrl, %commitCtrl, %SCSaveCtrl, %SCCommitCtrl, %SCBranchCtrl
-      = speculator[enable] %dataIn : !handshake.channel<i1>
+      = speculator[trigger] %dataIn : <[spec: i1]>, <i1, [spec: i1]>,
+        <i1, [spec: i1]>, <i1>, <i1>, <i3>, <i3>, <i1>
     ```
   }];
 
-  let arguments = (ins HandshakeType:$dataIn, ControlType:$enable);
+  let arguments = (ins HandshakeType:$dataIn, ControlType:$trigger);
   let results = (outs HandshakeType:$dataOut,
```

Comment:
Not much related to the main point here, but I think we should limit `dataIn` and `dataOut` to `ChannelType`

Reply:
done!